File: app/src/main/java/com/example/muzik/adapter/ListPlaylistAdapter.java
package com.example.muzik.adapter;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.muzik.response_model.PlayList;

import java.util.List;

public class ListPlaylistAdapter extends RecyclerView.Adapter<ListSongAdapter.ViewHolder>{
    private List<PlayList> playLists;

    @NonNull
    @Override
    public ListSongAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull ListSongAdapter.ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }
}

File: app/src/main/java/com/example/muzik/adapter/ListPlaylistAdapter.kt
package com.example.muzik.adapter;

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.muzik.R
import com.example.muzik.response_model.Playlist
import com.squareup.picasso.Picasso

class ListPlaylistAdapter(private val listPlaylist: List<Playlist>) : RecyclerView.Adapter<ListPlaylistAdapter.ListPlaylistViewHolder>() {

    class ListPlaylistViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListPlaylistViewHolder {
        return ListPlaylistViewHolder(
            LayoutInflater.from(parent.context).inflate(
                R.layout.item_playlist,
                parent,
                false
            )
        )
    }

    override fun getItemCount(): Int {
        return listPlaylist.size
    }

    override fun onBindViewHolder(holder: ListPlaylistViewHolder, position: Int) {
        val curPlaylist = listPlaylist[position]
        val tvPlaylistName: TextView = holder.itemView.findViewById(R.id.tvPlaylistsName)
        val playlistImage: ImageView = holder.itemView.findViewById(R.id.playlist_image_item)

        holder.itemView.apply {
            tvPlaylistName.text = curPlaylist.name

            if (curPlaylist.imageURL.isNotEmpty()) {
                Picasso.get()
                    .load(curPlaylist.imageURL)
                    .fit()
                    .centerInside()
                    .into(playlistImage)
            }
        }

    }


}
